### PR TITLE
modified freelist_hmap/hashmapGetFreePageIDs with better performance

### PR DESCRIPTION
**Pull request by cenkalti**
Resurrected #219

### CODE REVIEW

File: freelist_hmap.go [hunk 1]
```diff
@@ -80,12 +80,21 @@ func (f *freelist) hashmapGetFreePageIDs() []common.Pgid {
...
+		keys = append(keys, k)
+	}
+	sort.Sort(common.Pgids(keys))
+
+	for _, start := range keys {
+		size, ok := f.forwardMap[start]
+		if ok {
+			for i := 0; i < int(size); i++ {
+				m = append(m, start+common.Pgid(i))
+			}
```
Comment (ahrtr) [inline freelist_hmap.go:95]: Please consider to address the two minor comments in previous PR.

Comment (cenkalti) [reply freelist_hmap.go:94]: Done.

File: freelist_test.go [hunk 2]
```diff
@@ -434,3 +434,27 @@ func newTestFreelist() *freelist {
 
 	return newFreelist(freelistType)
 }
+
+func Test_freelist_hashmapGetFreePageIDs(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (cenkalti) [reply freelist_test.go:438]:
```
% TEST_FREELIST_TYPE=hashmap go test -v -run=XXX -bench=Benchmark_freelist -count=10 > old.txt
% git co freelist_hmap
% TEST_FREELIST_TYPE=hashmap go test -v -run=XXX -bench=Benchmark_freelist -count=10 > new.txt
% benchcmp old.txt new.txt
benchcmp is deprecated in favor of benchstat: https://pkg.go.dev/golang.org/x/perf/cmd/benchstat
benchmark                                       old ns/op     new ns/op     delta
Benchmark_freelist_hashmapGetFreePageIDs-12     5965148       144971        -97.57%
Benchmark_freelist_hashmapGetFreePageIDs-12     5993442       139366        -97.67%
Benchmark_freelist_hashmapGetFreePageIDs-12     5886753       150729        -97.44%
Benchmark_freelist_hashmapGetFreePageIDs-12     6057294       151722        -97.50%
Benchmark_freelist_hashmapGetFreePageIDs-12     5953581       164821        -97.23%
Benchmark_freelist_hashmapGetFreePageIDs-12     5922032       139932        -97.64%
Benchmark_freelist_hashmapGetFreePageIDs-12     12475581      144228        -98.84%
Benchmark_freelist_hashmapGetFreePageIDs-12     20996873      149522        -99.29%
Benchmark_freelist_hashmapGetFreePageIDs-12     9350295       163871        -98.25%
Benchmark_freelist_hashmapGetFreePageIDs-12     9285324       138735        -98.51%
```

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @cenkalti

**Review by serathius — APPROVED** [review]
99% time reduction for 1 alloc. Nice!

**Pull request merged**